### PR TITLE
Handle conflicting 'library' identifiers

### PR DESCRIPTION
**Pull request by Hixie**
Also, handle the new directory structure of the flutter repo.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @abarth for the directory structure
cc @devoncarew for the tool
cc @pq for the analyzer

**Comment by abarth** [comment]
Directory structure LGTM

File: packages/flutter_tools/lib/src/commands/analyze.dart [hunk 1]
```diff
@@ -16,6 +16,70 @@ import '../build_configuration.dart';
 import '../globals.dart';
 import '../runner/flutter_command.dart';
 
+bool _addPackage(String directoryPath, List<String> dartFiles, Set<String> pubSpecDirectories) {
+  bool foundOne = false;
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: This method could get a little simpler if you stored the length of dartFiles here, and compared that at the end to see if any files had been added.

File: packages/flutter_tools/lib/src/commands/analyze.dart [hunk 2]
```diff
@@ -16,6 +16,70 @@ import '../build_configuration.dart';
...
+  // .../directoryPath/*/bin/*.dart
+  // .../directoryPath/*/lib/main.dart
+  // .../directoryPath/*/test/*_test.dart
+  // .../directoryPath/*/test/*/*_test.dart
+  // .../directoryPath/*/benchmark/*/*_bench.dart
+
+  Directory binDirectory = new Directory(path.join(directoryPath, 'bin'));
+  if (binDirectory.existsSync()) {
+    for (FileSystemEntity subentry in binDirectory.listSync()) {
+      if (subentry is File && subentry.path.endsWith('.dart')) {
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: And a utility method, something like `_fileEndsWith(FileSystemEntity, String)`, would be able to shorten a bunch of these if statements. `bool _fileEndsWith(FileSystemEntity entity) => entity is File && entity.path.endsWith(ext)`

**Comment by devoncarew** [comment]
lgtm; 2 comments on existing code to try and reduce repetition.

File: packages/flutter_tools/lib/src/commands/analyze.dart [hunk 3]
```diff
@@ -331,6 +362,7 @@ linter:
     RegExp allowedIdentifiersPattern = new RegExp(r'_?([A-Z]|_+)\b');
     RegExp constructorTearOffsPattern = new RegExp('.+#.+// analyzer doesn\'t like constructor tear-offs');
     RegExp ignorePattern = new RegExp(r'// analyzer says "([^"]+)"');
+    RegExp conflictingNamesPattern = new RegExp('^The imported libraries \'([^\']+)\' and \'([^\']+)\' cannot have the same name \'([^\']+)\'\$');
```
Comment (pq) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: Maybe better would be to exclude the error via options.  Adding `import_duplicated_library_named` should do the trick.  That is, in `optionsBody`:

```
analyzer:
  errors:
    import_duplicated_library_named: ignore
    ...

```

PS: the error code name looks odd to me but it is what it is (defined in `error.dart`).

**Comment by pq** [comment]
% nit, LGTM.

File: packages/flutter_tools/lib/src/commands/analyze.dart (hunk 3 above)
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: We don't want to exclude the error, we just want to catch it and remove the temporary filename from the message.

Comment (pq) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: Ah, I see.  Over-zealous on my part. :+1:

File: packages/flutter_tools/lib/src/commands/analyze.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: done

File: packages/flutter_tools/lib/src/commands/analyze.dart (hunk 2 above)
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: I added three, isDartFile, isDartTestFile, and isDartBenchmarkFile, which seemed clearer and reduced the number of string literals.

**Comment by Hixie** [comment]
Ok, addressed comments. If travis goes green I'll check this in unless someone has further comments. :-)

**Pull request merged**